secrets/secrets_test.go

```diff
@@ -118,3 +118,65 @@ func TestNewSecrets(t *testing.T) {
 		})
 	}
 }
+
+func TestSecretsWrongType(t *testing.T) {
+	rawSecrets := `
+		{
+			"secrets": {
+				"secret/myservice/external-account-key": {
+					"type": "versioned",
+					"current": "YWJjZGVmZ2hpamtsbW5vcHFyc3R1dnd4eXowMTIzNDU=",
+					"previous": "aHVudGVyMg=="
+				},
+				"secret/myservice/some-api-key": {
+					"type": "simple",
+					"value": "Y2RvVXhNMVdsTXJma3BDaHRGZ0dPYkVGSg==",
+					"encoding": "base64"
+				},
+				"secret/myservice/some-database-credentials": {
+					"type": "credential",
+					"username": "spez",
+					"password": "hunter2"
+				}
+			},
+			"vault": {
+				"url": "vault.reddit.ue1.snooguts.net",
+				"token": "17213328-36d4-11e7-8459-525400f56d04"
+			}
+		}
+	`
+	buf := bytes.NewBuffer([]byte(rawSecrets))
+	secrets, err := NewSecrets(buf)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = secrets.GetSimpleSecret("secret/myservice/external-account-key")
+
+	if err == nil || err.Error() != "secrets: requested secret at path 'secret/myservice/external-account-key' of type 'simple' does not exist, but a secret of type 'versioned' does. Consider using the correct API to retrieve the secret" {
```

Comment:
while it's probably ok to have one test comparing `err.Error()` to make sure the text is up to expectation, for the majority of the tests, you should use `errors.As` and then compare the fields, over comparing the error text directly, in tests.

actually with the current implementation of `SecretWrongTypeError`, just use `errors.Is` in tests probably also works and is simpler.